SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -141,9 +140,12 @@ bool USpatialNetDriver::InitBase(bool bInitAsClient, FNetworkNotify* InNotify, c
 		return false;
 	}
 
+	//Init InterestFactory
+	SpatialInterestFactory = MakeUnique<SpatialGDK::InterestFactory>(ClassInfoManager);
+
 	if (!bInitAsClient)
```

Comment:
Maybe this should probably be passed as an argument to the constructor of the interest factory? then whether or not this gets created would be up to the factory

Reply:
I don't feel super strongly either way here. I believe you are suggesting adding a bool arg to the InterestFactory constructor that signifies whether we are on the client or not. My only slight issue with that is this. Atm everything that is done by the InterestFactory we explicitly tell it to do. Making the suggested change would move away from that in that the InterestFactory would reason for itself what it needs to do. The InterestFactory would then understand about clients and servers and how that affects it's own behavior. If in the InterestFactory we already do this kind of reasoning then it sounds like a good change.

Reply:
I do see some of the value in your other comment about this though. If there is an argument, the person creating the InterestFactory has to pass that arg. If he does he hopefully understands that arg. This ensures the CreateClientCheckoutRadiusConstraint can not be omitted.

Reply:
I think we currently allow the interest factory to decide what interest to create based on the actor that is passed in. However, I see what you mean- maybe a different question- why do we have an interest factory on clients at all? Should there be a factory that returns interest factories depending on whether it is a client or server? This way we could return a noop interest factory for clients, and not worry about doing any work there.